client/src/AttendanceView/AttendanceView.js

```diff
@@ -21,6 +23,53 @@ export function AttendanceView() {
   // columns will be current dates
   const { token } = useSelector(state => ({ token: state.auth.token }))
   const [dateSelected, setDateSelected] = useState(dayjs())
+  const [editAttendanceModalData, setEditAttendanceModalData] = useState(null)
+  const [updatedAttendanceData, setUpdatedAttendanceData] = useState([{}, {}])
+  const [modalButtonDisabled, setModalButtonDisabled] = useState(false)
+  const latestAttendanceData = useRef(updatedAttendanceData)
+  const titleData = useRef({ childName: null, columnDate: null })
+
+  const updateAttendanceData = data => {
```

Comment:
Can this and the modal logic go into its own component?  We may not reuse it but it might make sense to keep this logic out of the View and just pass the data down for editing.  There's a LOT of modal logic in here and the file is getting pretty large and multi-responsibility.

Reply:
yes! I am actually working on this now! also helps for testing!